File: 0128_Longest_Consective_Sequence/0128_Longest_Consective_Sequence.cc
#include <unordered_set>
#include <vector>
#include <algorithm>>
class Solution {
public:
    int longestConsecutive(std::vector<int>& nums) {
        std::unordered_set<int> set(nums.begin(), nums.end());

        int length = 0;
        int max_length = 0;
        int x = 0;
        for (auto n : nums) {
            if (!set.count(n-1)) {
                length = 0;
                x = n;
                while (set.count(x)) {
                    x += 1;
                    length += 1;
                }
                max_length = std::max(length, max_length);
            }
        }
        return max_length;
    }
};

File: 0128_Longest_Consective_Sequence/0128_Longest_Consective_Sequence.py
def longestConsecutive(self, nums: List[int]) -> int:
    hashset = set(nums)
    i, length, max_length = 0, 0, 0
    for i in range(0, len(nums)):
        n = nums[i]
        length = 0
        if n-1 not in hashset:
            while n in hashset:
                n += 1
                length += 1
            max_length = max(length, max_length)
    return max_length